src/main/java/com/sergiofah/integracao/controller/ProductPageController.java

```diff
@@ -46,57 +50,81 @@ public class ProductPageController {
 	@FXML
 	private ImageView productImageView;
 
-    public ProductPageController() {
+	private Image loading;
+
+	private Long selectedLineId;
+
+	private List<LineDTO> lineDTOList;
+	private List<CategoryDTO> categoryDTOList;
+	private List<ProductDTO> productDTOList = new ArrayList<>();
+
+	public ProductPageController() {
         loading = new Image(getClass().getResourceAsStream("/images/loading.gif"));
     }
 
     @FXML
 	private void initialize() {
-		productDAO = new ProductDAO();
-		populateComboBox();
+		Service service = new Service();
+		LineService lineService = new LineService();
+		if (service.getServerStatus()) {
+			this.lineDTOList = lineService.getLines();
+			populateComboBox();
+		} else {
+			serverStatus.setText("Status: Server Offline");
+			serverStatus.setTextFill(Color.RED);
+		}
 	}
 
-	public void populateComboBox(){
-		linesComboBox.setItems(FXCollections.observableArrayList(productDAO.getLines()));
+	public void populateComboBox() {
+		linesComboBox.setItems(FXCollections.observableArrayList(lineDTOList.stream().map(LineDTO::getLine).collect(Collectors.toList())));
 	}
 	
 	@FXML
 	private void OnClickComboBox() {
-		selectedLine = linesComboBox.getValue();
+		selectedLineId = lineDTOList.stream()
+				.filter(id -> id.getLine().equals(linesComboBox.getValue()))
+				.map(LineDTO::getId)
+				.findFirst()
+				.get();
 		populateTreeView();
 		modelsTitledPane.setDisable(false);
 		modelsTitledPane.setExpanded(true);
 	}
 	
 	private void populateTreeView() {
-		
-    	List<Category> categoryList = productDAO.getCategoriesFromLine(selectedLine);
-
-        TreeItem<String> rootItem = new TreeItem<>(selectedLine);
+		CategoryService categoryService = new CategoryService();
+		ProductService productService = new ProductService();
```

Comment:
All services objects could be global attributos but productService must be because it is used not only in `populateTreeView()` method but in `selectionTreeViewHandler()` as well.